src/NServiceBus.Core/Pipeline/RegisterStep.cs

```diff
@@ -48,14 +48,14 @@ protected RegisterStep(string stepId, Type behavior, string description, Func<IS
         /// </summary>
         public Type BehaviorType { get; private set; }
 
-        internal void ApplyContainerRegistration(IServiceCollection container)
+        internal void ApplyContainerRegistration(IServiceCollection serviceCollection)
         {
             if (factoryMethod != null)
             {
                 return;
             }
 
-            container.ConfigureComponent(BehaviorType, DependencyLifecycle.InstancePerCall);
+            serviceCollection.AddWithInterfaces(BehaviorType, ServiceLifetime.Transient);
```

Comment:
Making this just AddTransient only broke the DI signature test since

```
NServiceBus.Pipeline.IBehavior`2[[NServiceBus.Pipeline.IIncomingLogicalMessageContext, NServiceBus.Core, Version=9.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c],[NServiceBus.Pipeline.IIncomingLogicalMessageContext, NServiceBus.Core, Version=9.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c]] - Transient
NServiceBus.Pipeline.IBehavior`2[[NServiceBus.Pipeline.IIncomingPhysicalMessageContext, NServiceBus.Core, Version=9.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c],[NServiceBus.Pipeline.IIncomingPhysicalMessageContext, NServiceBus.Core, Version=9.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c]] - Transient
```

was removed but all other tests still passed